odin_infrastructure/odin_api_stack.py

```diff
@@ -0,0 +1,78 @@
+from aws_cdk import Stack, aws_route53
+from aws_cdk import aws_ec2 as ec2
+from constructs import Construct
+
+from .admin_host import AdminInstance
+from .config import ODIN_API_EIP
+from .mongo import MongoInstance
+from .odin_cluster import OdinService
+
+
+class OdinAPIStack(Stack):
+    def __init__(self, scope: Construct, id: str, **kwargs) -> None:
+        super().__init__(scope, id, **kwargs)
+
+        nat_gateway_provider = ec2.NatProvider.instance(
+            instance_type=ec2.InstanceType("t3.small")
+        )
+        vpc: ec2.IVpc = ec2.Vpc(
+            self,
+            "OdinVPC",
+            nat_gateway_provider=nat_gateway_provider,
+            max_azs=2,
+            nat_gateways=1,
+            vpc_name="OdinVPC",
```

Comment:
Is this name all other services need to know, for their stacks to set up permissions etc?

Reply:
Yes, unless that service need access to a bucket or so.